.github/workflows/test-runner-nightly-perf.yml

```diff
@@ -0,0 +1,54 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Sample Nightly Perf Tests
+
+on:
+  workflow_dispatch: {}
+  schedule:
+    - cron: '0 14 * * *' # run at 2 PM UTC, 9 AM EST.
+
+jobs:
+  test-runner-integration:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up AdoptOpenJDK 11
+        uses: joschi/setup-jdk@v2
+        with:
+          java-version: 11
+      - name: Get Vault token
+        id: vault-token-step
+        env:
+          VAULT_ADDR: https://clotho.broadinstitute.org:8200
+        run: |
+          VAULT_TOKEN=$(docker run --rm --cap-add IPC_LOCK \
+            -e "VAULT_ADDR=${VAULT_ADDR}" \
+            vault:1.1.0 \
+            vault write -field token \
+              auth/approle/login role_id=${{ secrets.VAULT_APPROLE_ROLE_ID }} \
+              secret_id=${{ secrets.VAULT_APPROLE_SECRET_ID }})
+          echo ::set-output name=vault-token::$VAULT_TOKEN
```

Comment:
Why `echo` without having something receiving the output?

Reply:
`echo ::set-output...` and `echo ::add-mask...` are commands that get read by the Github runner itself, [see docs](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions). `set-output` makes output available for other steps, and `add-mask` prevents the specified value from being logged.

Reply:
According to documentation [Workflow commands for GutHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#using-workflow-commands-to-access-toolkit-functions).

The `echo` command is used to send the workflow command (in this case `set-output`) to the runner over `stdout`, which is equivalent to executing an API `core.setOutput('vault-token', $VAULT_TOKEN)`. So it's not using `echo` is the usual sense.